http/middleware.go

```diff
@@ -0,0 +1,150 @@
+// Package http provides HTTP utilities and handler middleware.
+package http
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+	jwt "github.com/clerk/clerk-sdk-go/v2/jwt"
+)
+
+// RequireAuthorization attempts to get Clerk authorization claims
+// from the http.Request context and responds with status  403 Forbidden
+// if it fails to verify the claims.
+func RequireHeaderAuthorization(ctx context.Context, opts ...AuthorizationOption) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return WithHeaderAuthorization(ctx, opts...)(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			claims, ok := clerk.SessionClaimsFromContext(r.Context())
```

Comment:
Good question, I had the same doubts.

 I left the `ctx` context for possible future usage. 

Although it's not actually needed, if we don't add it now and we need it in a future version, we'll break the API.

Do you think it's better to remove it?

Reply:
I'm not sure I can think of a good usage pattern here too, since the middleware setup doesn't happen within an operation context. Perhaps we should remove it?

Reply:
Removed the context from the arguments.